index.ts

```diff
@@ -71,9 +71,6 @@ async function togglePermission(tab: chrome.tabs.Tab, toggle: boolean): Promise<
 
 	const userAccepted = await chromeP.permissions.request(permissionData);
 	if (!userAccepted) {
-		chrome.contextMenus.update(contextMenuId, {
-			checked: false,
-		});
 		return;
```

Comment:
I still don't understand what's the problem with this line.

1. item is unchecked
2. user clicks it
3. chrome checks it
4. we ask for permission
5. user denies ❌ 
6. this piece of code is reached and we uncheck it
7. done, no need for `finally`

Or if successful:

1. item is unchecked
2. user clicks it
3. chrome checks it
4. we ask for permission
5. user accepts ✅ 
6. this piece of code is NOT reached
7. done, no need for `finally` wither

I can't think of another scenario in which `request()` returns `true` but it should not be (natively) checked.

To put it in other words, we only have to fix the check:

- if the user rejects
- if it throws

The way I see it, the only reason https://github.com/fregante/webext-domain-permission-toggle/issues/29 is an issue is because `updateItem()` is called even if successful. That's it.

Reply:
> I can't think of another scenario in which request() returns true but it should not be (natively) checked.

I've been through so many scenarios now that my head is exploding, but I'll do my best to respond accurately.

I don't think I ever said there was a problem with these three lines; it's just that my PR proposed another way of tackling the issue via modified usage of `updateItem()` elsewhere, which made these three lines redundant (since we don't want to be updating the same menu item twice in rapid succession as part of the same event handling).

The problematic situation I described [above](https://github.com/fregante/webext-domain-permission-toggle/pull/33#discussion_r1460673293) was not relating to when the user wants to grant permission, but rather when they are trying to remove it:

1. item is checked and in the manifest but not disabled (i.e. greyed out) 
2. user clicks it
3. we ask to remove permission
4. browser throws an exception because the URL matches a pattern already in the manifest
5. at this point we need to make sure that the checkmark stays present

Yes it's true that in step 1 here, the item should be disabled, but as I observed above, this is not yet working reliably in all situations, which is why I was proposing a less brittle approach which doesn't assume that everything else is working perfectly.

But anyway, I think your draft approach in #38 is more promising so hopefully that will work well in which case I'm happy to leave this PR closed.

Reply:
> browser throws an exception because the URL matches a pattern already in the manifest

This is where we keep butting heads. This does not happen **if you keep the Chrome UI** out of the equation.

Reply:
> doesn't assume that everything else is working perfectly.

The issue with that is that you're already working on a broken piece of UI, so what you're suggesting is focusing on the wrong part of the code. Let's instead make sure that the UI is correct *before* the click.

Reply:
Sure, I've already given up worrying about the Chrome UI for now.  However there are still problems, but I'll describe them in the relevant issues and PRs.